posts/hey-this-looks-familiar-front-end-development-for-back-end-engineers.md

```diff
@@ -0,0 +1,571 @@
+
+``` js
+updateInnerHTML() {
+  this.divElement.innerHTML = this.getInnerHTML();
+}
+
+updateClass() {
+  this.divElement.setAttribute('class', this.model.colorClass);
+}
+```
+
+Now, we're ready to pass in values for these attributes from our `index.html` document. We'll do the DC card both ways so we can compare. 
+
+``` html
+<weather-card-component></weather-card-component>
+<weather-card-component
+  location="Washington, District of Columbia"
+  temp="30"
+  condition="partly cloudy"
+  updated="2022-01-31 20:30"
+  color_class="purple"
+></weather-card-component>
+<div class="weather-card purple">
+  <h3>In Washington, District of Columbia,<br/>it is currently 30&deg;F and partly cloudy.</h3>
+  <p>Last updated: 2022-01-31 20:30</p>
+</div>
+```
+
+We see that we have two identical "cards" for DC, which is what we expected.
+
+![Page output after adding the web component for Washington DC, with identical cards created by the web component and previously-existing HTML.](../images/2022/02/fed4bees2-wdc-component.png)
+
+Notice that we did't need to pass in any attributes for the Richmond office, because the model in our constructor provides those values as defaults. While that's fine for this current page, it's not what someone might expect from an "empty" component outisde of our context. Let's provide some default text to display when information is missing.
+
+``` js
+that.model = {
+  location: "an unknown location",
+  tempFahrenheit: "an unknown temperature",
+  weatherCondition: "unknown conditions",
+  lastUpdated: "unknown",
+  colorClass: null
+};
+```
+
+In our `index.html` document, we'll update our Richmond card to pass in the data using attribute binding and add a custom element with no attributes set. 
+
+![Page output after adding the default text to show when no attribute is passed in. The no-attribute card has a white background, and the sentence describing the weather for that card is "In an unknown location, it is currently an unknown temperature°F and unknown conditions.". ](../images/2022/02/fed4bees2-default-text.png)
+
+We've reached something that is _almost_ reasonable for an "empty" component. It would be better to not have the `°F` if we don't know the temperature. We'll get to that, but first, let's take a look at how we wrote `attributeChangedCallback` and how we might refactor it.
+
+### Climbing Up On My "Refactoring for Readability" Soapbox
+
+<!-- I added this section based on some feedback from Cody on clean/clear use of JavaScript. I can edit to be less opinionated and more concise, but I would prefer to keep the original code as it is and address the refactor. I think optimizing for understanding the distinction between attributes/fields is important when I first introduce attribute binding. I did incorporate Cody's suggestions into the props section below without first doing it my weird way (lol). -->
```

Comment:
I saw the mention in the props section too, so I think you could cut that part here if you don't want to do both examples and continue on with the flow of the article. Whatever you prefer.

Reply:
Okie. I kindof like this section, so I think I'll leave as is.